gateway-js/src/index.ts

```diff
@@ -528,20 +613,49 @@ export class ApolloGateway implements GraphQLService {
     }
   }
 
-  private async updateWithSupergraphSdl(
-    result: SupergraphSdlUpdate,
-  ): Promise<void> {
-    if (result.id === this.compositionId) {
-      this.logger.debug('No change in composition since last check.');
-      return;
+  private externalSupergraphUpdateCallback(supergraphSdl: string) {
```

Comment:
Can we check that this isn't called before the phase is loaded (ie that it isn't called during the initial supergraphSdl function invocation)?

Reply:
It's actually not possible (the hook isn't called until the gateway successfully loads, so the user has no handle on the function yet). I did address a couple other states the gateway could be in though: c2a88ea

Reply:
I went the switch route with a default case. Also helped me realize that calling this from `initialized` is also valid :)